python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -87,14 +91,16 @@ def __init__(
             # TODO(ekl): require config reader to be a callable always.
             config_reader: Union[str, Callable[[], dict]],
             load_metrics: LoadMetrics,
+            gcs_node_info_stub: gcs_service_pb2_grpc.NodeInfoGcsServiceStub,
```

Comment:
how much work would it be to move this to monitor?
maybe file an issue for tracking?

Reply:
Along the lines of what you've proposed, an autoscaler update should produce a single scale recommendation consisting of who to terminated and how many of each node type to launch.

Once we have things refactored to work that way, we can have monitor.py act on that the recommendation -- acting would include draining nodes and in some circumstances forwarding the scale request to an external operator.

Reply:
Will file an issue.

Reply:
https://github.com/ray-project/ray/issues/20240